### PR TITLE
Fixing initialization declarations

### PR DESCRIPTION
**Pull request by weverb2**
There were some Kotlin related issues that I'll explain with line comments

### CODE REVIEW

File: app/templates/screen/Presenter.kt.mustache [hunk 1]
```diff
@@ -36,14 +36,14 @@ class {{screenUpperCamel}}Presenter(private val context: Context) {
     }
 
     fun onResume() {
-        subscriptions = RxUtils.getNewCompositeSubIfUnsubscribed(subscriptions)
```
Comment (weverb2) [inline app/templates/screen/Presenter.kt.mustache]: This caused a lateinit exception. 

I may be wrong, but I don't see a reason to not just create a new sub here.

File: app/templates/screen/Presenter.kt.mustache [hunk 2]
```diff
@@ -36,14 +36,14 @@ class {{screenUpperCamel}}Presenter(private val context: Context) {
...
+        subscriptions = CompositeSubscription()
         if (!state.initialized) {
             state.initialized = true
         }
     }
 
     fun onPause() {
-        RxUtils.unsubscribeIfNotNull(subscriptions)
```
Comment (weverb2) [inline app/templates/screen/Presenter.kt.mustache]: Non nullable subscription means we don't need a util call here. (Even if we did I'd prefer a `?` call anyway)

File: app/templates/screen/Presenter.kt.mustache [hunk 3]
```diff
@@ -18,7 +18,7 @@ class {{screenUpperCamel}}Presenter(private val context: Context) {
     }
 
     private lateinit var subscriptions: CompositeSubscription
-    private var view: {{screenUpperCamel}}ViewContract? = null
```
Comment (weverb2) [inline app/templates/screen/Presenter.kt.mustache]: There is no reason for ViewContracts to be nullable. They are never set back to null.

**Comment by patrickhammond** [comment]
Future PR would be to remove `RxUtils` entirely.  I think we might have outgrown its usefulness.

**Pull request merged**